specification/attributes/discount_handling.md

```diff
@@ -0,0 +1,46 @@
+
+Some of the discount offers can be purchased from a provider in order to get reduced prices. The most common example of this is commitment-based discounts, where you "purchase" a commitment to use or spend a specific amount within a period. There are instances where a portion of the purchased discount may go unused. This can result in effectively lost savings and needs to be clearly identifiable at a granular level to enable FinOps scenarios. In order to facilitate this, the amount of a purchased discount is amortized over the term that the discount is applied to (e.g., 1 year). Amortization is a process used to break down and spread the purchase costs over a period of time or term of use. When a purchase is applicable to resources, like commitment-based discounts, the amortized cost of a resource takes the initial payment and term into account and distributes it out based on the resource's usage, attributing the prorated cost for each hour of billing. Amortization provides a method for FinOps practitioners to reallocate purchase charges to the appropriate audience in support of their cost allocation efforts. Discount Handling for purchased commitments is commonly used for scenarios like calculating utilization and implementing chargeback of the purchase amount.
+
+While providers may use different terms to describe discounts, FOCUS identifies a discount as being a reduced price applied directly to a row or charge. Any price or cost reductions that are awarded after the fact are identified as a "Credit" Charge Subcategory. One example might be when a provider offers a reduced rate after passing a certain threshold of usage or spend.
+
+All rows defined in FOCUS MUST follow the discount handling requirements listed below.
+
+## Attribute ID
+
+DiscountHandling
+
+## Attribute name
+
+Discount Handling
+
+## Description
+
+Indicates how to include and apply discounts to usage charges or rows. 
+
+## Requirements
+
+* All applicable discounts MUST be applied to each row they pertain to and MUST NOT be negated in a separate row.
+* All discounts applied to a row MUST apply to the entire charge. 
+  * If a discount only applies to a subset of the charge, then that portion of the charge MUST be split into a separate row. One with the discount and one without.
+  * If multiple discounts apply, there MUST be separate rows for each unique combination of applied discounts or undiscounted amount.
+  * Each discount MUST be identifiable using existing FOCUS columns. 
+    * Discounts applied to charges/rows from a commitment-based discount MUST include a CommitmentDiscountId.
+    * If a provider applies a discount that cannot be represented by a FOCUS column, they SHOULD include additional columns to identify the source of the discount.
```

Comment:
Do we consider spot instance usage as a form of discount? For GCP SUD, we could consider the discount row as a General Adjustment -> Credit type but spot usage is associated with a resource but with a discounted price; and no CommitmentDiscountId. Or we just consider it as a normal Usage row but with a different (lower) price list, meaning, not discounted?

> If a provider applies a discount that cannot be represented by a FOCUS column, they SHOULD include additional columns to identify the source of the discount.
I was thinking that spot instance usages would fall under this provision but this line seems a bit open-ended. Specifically the "additional columns" part, which could potentially be handled post v1.0?

Reply:
IMHO, spot is not strictly a discount, because spot instances have a different SLA than regular instances: their usage can be interrupted. I see more spot as "low-cost" than "discount".

Reply:
yeah, I'd view spot as a lower cost vm.

Reply:
I feel like it's a bit unclear what we're asking providers to do. Are we saying additional provider-defined columns should be included to identify the discount details?